### PR TITLE
Adds a config toggle for granting admin server profile access

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
It only took 6 months, but someone has finally taken the 60 seconds to add a config entry to enable/disable admin access to the profiler. 

Closes: #60583

## Why It's Good For The Game
Kills a 6 month old testmerge PR

## Changelog

:cl: Ryll/Shaps
config: Added ALLOW_ADMIN_PROFILING entry to the config, controlling whether admins are given access to the server profiler
/:cl:

### CODE REVIEW

File: config/config.txt [hunk 1]
```diff
@@ -599,3 +599,6 @@ CACHE_ASSETS 0
 ## The number of unique PR announcements allowed per round
 ## This makes sure that a single person can only spam 3 reopens and 3 closes before being ignored
 PR_ANNOUNCEMENTS_PER_ROUND 5
+
+## When enabled, grants those with R_DEBUG permissions to use the server profile debug functionality
+ALLOW_ADMIN_PROFILING 1
```
Comment (Mothblocks) [inline config/config.txt:604]:
```suggestion
ALLOW_ADMIN_PROFILING
```

If it's a flag, you don't need `1`.

**Comment by MrStonedOne** [comment]
So, the hidden secret i've never exposed, is that that pr isn't even fixing the issue.

This config needs to exist, but a second one needs to exist as well, and thats a config to disable all profiling by overriding the world.profile proc and returning at the top (has to be this way).

See, doing most world.profile actions while a profile is not going, will in fact start one.

So the lag the test merge was trying to fix? ya it keeps coming back as long as the time track, profiler, or server maintenance subsystems are running.

File: config/config.txt [hunk 2]
```diff
@@ -599,3 +599,6 @@ CACHE_ASSETS 0
 ## The number of unique PR announcements allowed per round
 ## This makes sure that a single person can only spam 3 reopens and 3 closes before being ignored
 PR_ANNOUNCEMENTS_PER_ROUND 5
+
+## When enabled, grants those with R_DEBUG permissions to use the server profile debug functionality
+ALLOW_ADMIN_PROFILING
```
Comment (MrStonedOne) [inline config/config.txt:604]: invert this. If the config doesn't exist (ie, existing configs) it should retain old behavior.

changing the default in code means you have to do `allow_admin_profiling 0` to disable it which is lame, changing it to forbid_admin_profiling would be less lame.

**Comment by LemonInTheDark** [comment]
> So, the hidden secret i've never exposed, is that that pr isn't even fixing the issue.
> 
> This config needs to exist, but a second one needs to exist as well, and thats a config to disable all profiling by overriding the world.profile proc and returning at the top (has to be this way).
> 
> See, doing most world.profile actions while a profile is not going, will in fact start one.
> 
> So the lag the test merge was trying to fix? ya it keeps coming back as long as the time track, profiler, or server maintenance subsystems are running.

cursed

File: code/game/world.dm [hunk 3]
```diff
@@ -374,5 +374,9 @@ GLOBAL_VAR(restart_counter)
 /world/proc/on_tickrate_change()
 	SStimer?.reset_buckets()
 
+/world/proc/Profile(command, type, format)
```
[the remark this answers is not in the document]
Comment (Ryll-Ryll) [reply code/game/world.dm:377]: blah i didn't realize i actually pushed that commit, i think i just did it so switch branches

**Comment by Ryll-Ryll** [comment]
Should be good

**Comment by Ryll-Ryll** [comment]
@MrStonedOne we did it team

**Comment by MrStonedOne** [comment]
not quite, it allows the world start profiler start, but blocks the stop, so it keeps going.

You'll have to check the the command and also allow it if it has the `PROFILE_STOP` bitflag

File: code/game/world.dm [hunk 4]
```diff
@@ -379,5 +379,9 @@ GLOBAL_VAR(restart_counter)
 /world/proc/on_tickrate_change()
 	SStimer?.reset_buckets()
 
+/world/Profile(command, type, format)
+	if(!global.config?.loaded || !CONFIG_GET(flag/forbid_all_profiling))
```
Comment (MrStonedOne) [inline code/game/world.dm:383]:
```suggestion
	if((command & PROFILE_STOP) || !global.config?.loaded || !CONFIG_GET(flag/forbid_all_profiling))
```

Comment (Ryll-Ryll) [reply code/game/world.dm:383]: f

**Comment by Ryll-Ryll** [comment]
I assume you know what you're doing with that, but it also makes sense to allow stops and not starts

File: config/config.txt [hunk 5]
```diff
@@ -602,3 +602,9 @@ CACHE_ASSETS 0
 ## The number of PR announcements allowed for the same pull request per round
 ## This makes sure that somebody can only spam a few reopens and closes before being ignored
 PR_ANNOUNCEMENTS_PER_ROUND 5
+
+## Uncomment to block all attempts to profile, for performance reasons
+#FORBID_ALL_PROFILING
```
Comment (MrStonedOne) [inline config/config.txt:607]:
```suggestion
## Uncomment to block all attempts to profile, for performance reasons
##   Note, this option will not override the "AUTO_PROFILE" config.
#FORBID_ALL_PROFILING
```

optional, but useful docs. im gonna reboot campbell and tell you if everything is working properly.

**Review by MrsTonedOne — APPROVED** [review]

**Pull request merged**